### PR TITLE
Menu screens updated

### PR DESCRIPTION
**Pull request by yds05238**
## Overview

Updated menu screens following updated designs. Added new pages (game settings, name menu, popup loading/error window, enter name menu). Added game settings and player ready notification across network. Need testing (especially game lobby networking with 2+ players). 

## Changes Made

- Updated loading scene (removed transitions)
- Updated join room menu (5 digit room id with number pad input)
- Changed main menu button layouts
- Removed parallax effect from player settings 
- Updated game lobby menu (changed asset layout, added networking for game settings and player ready)
- Added game settings menu (moved all game settings logic and assets here)
- Added popup menu (handle loading over network on creating/joining/reconnecting game, display loading/error messages)
- Added name menu (first time player check using saved settings file)
- Removed unused methods from menu scene
- Refactored Network Message Manager

## Next Steps (delete if not applicable)

- Update visuals with the finalized menu designs and textures 
- Currently there is a bug in game lobby network disconnect handling (host does not return to main menu when in game settings) (network reconnecting/disconnect is slow from game lobby and game settings)

## Screenshots (delete if not applicable)

- Main Menu

mainmenu

- Create Game Loading 

creategame_loading

- Create Game Error

creategame_error

- Settings Menu (player settings)

settings

- Enter Name Menu (first time only)

namemenu

- Join Menu 

joinmenu

- Join Menu (Loading)

joinmenu_loading

- Join Menu (Error)

joinmenu_error

- Game Lobby (Non-Host)

lobby_nonhost

- Game Lobby (Non-Host, Ready)

lobby_nonhost_ready

- Game Lobby (Host)

lobby_host

- Game Lobby (Reconnecting)

lobby_reconnect

- Game Lobby (Error)

lobby_error

- Game Setting Menu (Host)

gamesettings_host

- Game Setting Menu (Non-Host)

gamesettings_nonhost

- Game Setting Menu (Non-Host, Reconnecting)

gamesettings_nonhost_reconnecting

- Game Setting Menu (Non-Host, Error)

gamesettings_nonhost_error

### CODE REVIEW

**Review by kevinsun-dev — APPROVED** [review]
I didn't look through the code that closely, since it's all roughly about right for what we're doing. What I did do was run it, and see if it crashed or failed in unexpected ways. That it didn't do, so I'm happy with the results. 🚀 

@henryz2000 Can you pull the branch and make sure it works for Windows?
@steiner26 Check if this branch works correctly with your music. I couldn't drag the settings bar.

**Comment by wlong123** [comment]
Looks good! Only things I noticed are this log: ERROR: Dimensions for ninepatch 'patch' are too small and 
when a player unsuccessfully creates a new game the error popup does not go away and it attempts to create a new game again. I think the popup should just disappear

File: source/CIPopupMenu.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+                _networkMessageManager->receiveMessages();
+
+                if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Reconnecting) {
+                    break;
+                }
+                else if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Disconnected) {
+                    _timer = 2.4f;
+                    break;
+                }
+                else if (_networkMessageManager->getNetworkStatus() == CUNetworkConnection::NetStatus::Connected) {
```
Comment (wlong123) [inline source/CIPopupMenu.cpp:218]: I think you need a case for CUNetworkConnection::NetStatus::GenericError. This should cause it to not try to recreate a game after failing to create a game

**Comment by henryz2000** [comment]
added files to vs

**Pull request merged**